spock-core/src/main/java/org/spockframework/util/RenderUtil.java

```diff
@@ -52,7 +53,10 @@ public static String toStringOrDump(@Nullable Object value) {
       Wrapper wrapper = (Wrapper) value;
       return String.format("%s as %s", toStringOrDump(wrapper.unwrap()), wrapper.getType().getSimpleName());
     }
-    return dump(value);
+    if (value.getClass().isArray()) {
+      return dumpArrayString((Object[]) value, RenderUtil::toStringOrDump);
```

Comment:
💬 Maybe it does not matter whether `toStringOrDump` or `inspectOrDump` is called to render array arguments. But I decided to be on a safer side.